### PR TITLE
Add API

### PR DESCRIPTION
**Pull request by SaboorSohaib**
## Hi Dear :wave: 

**In this project, I added the following requirements.**

- Create a new Rails API app called 'hello-rails-back-end'.
- Initialize your project with Git.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).

- Create a new React app called 'hello-react-front-end'
- Initialize your project with Git.
- Create the App component with react-router.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

:diamond_shape_with_a_dot_inside: [Front End Pull Request Link](https://github.com/SaboorSohaib/hello-react-front-end/pull/1)

**Thank you.**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,172 @@
...
+
+If you like this project you can fork it.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank... <a href="https://www.microverse.org/?gclid=CjwKCAiArY2fBhB9EiwAWqHK6s-2-x4d57Pghz47XT1BgsYuF81ZprM-k-IwzI0_L96nV0SQ93A8ExoCVnQQAvD_BwE" title="planet icons">Microvers for giving me this opportunity to build this project</a>
```
Comment (Davidosky007) [inline README.md:166]: * Kindly add the license section of your readme 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well-written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
* I can see you have not added CORS configuration to your backend app this enables cross-origin resource sharing between your frontend and backend app 💭 YOU can go to the cors.rb file and uncomment this default config in the screenshot below 👇and install the cors gem
image

image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1,24 +1,180 @@
...
+- We can see random greeitngs.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- <a href="https://github.com/SaboorSohaib/hello-react-front-end" target="_blank">React Application</a>
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (BenMukebo) [inline README.md]: - [OPTIONAL] This should not be inside the `Live Demo` section, 🤔 The live demo section is supposed to contain the application deployment link, please correct this 💯 👍🏻 
- Note: you can remove the Live demo section if you don't have a deployment link but you need to keep the link to your Frontend repo inside the README file.

File: db/migrate/20230502113352_create_message.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateMessage < ActiveRecord::Migration[7.0]
+  def change
+    create_table :messages do |t|
+      t.text :greeting
+
```
Comment (BenMukebo) [inline db/migrate/20230502113352_create_message.rb]: - [ ] Great job so far 👏🏻 However the database table for storing different greetings should be created correctly. I can see that you have created the message migration but you should also have created the `message model`. please address this. 👍🏻 

- [ ] Kindly make sure you can correctly Create 5 different greetings using the rails console or seed. 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @SaboorSohaib, 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉
- Git flow is followed. :heavy_check_mark:
- The route is created. ✔️ 
- The controller with one action was created. :heavy_check_mark:
- PR has a good title and summary. ✔️

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

- [OPTIONAL ] Use descriptive commit messages.  The commit message should be written in the imperative tense and `capitalized`. [Read more about how to write descriptive commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8). Example:
  - :x: Wrong: added Descriptive README
  - :heavy_check_mark: Right: Add descriptive README
 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/message.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class Message < ApplicationRecord
+end
```
Comment (Munsa1) [inline app/models/message.rb:2]: - [ ] Good job adding the `message` model, the user can now create a greeting and be able to view it on the back end.
image
However, you have to ensure the front-end is able to display the greetings. Currently when I run the front end, no message is being displayed.
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Descriptive Pr :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] You could consider adding the MIT license agreement to the front end readme file

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/message.rb (hunk 4 above)
Comment (KDlamini) [reply app/models/message.rb:2]: Well resolved ✔️ 

image

**Review by KDlamini — APPROVED** [review]
### Hi @SaboorSohaib 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Professional Pull request
✔️ Model, Controllers, Migrations, and Routes for Messages are set up correctly
✔️ API endpoints return random Greeting
✔️ API uses cors to fetch and render Greetings in UI

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**